### ISSUE TITLE
Real implementation !!!

### ISSUE THREAD

**Issue by DerXanRam** [jrbjrb1212/Cpp-Crypto-Arbitrage-Framework#3]
I have been trying for a week to find a way about how to implement cross exchange arbitrage. Arbitrage only in one exchange is very hard because it results loss due to it's not that hard for the exchange to maintain it's liquidity and arbitrage gap.  However, when u try to arbitrage between 2 or more exchanges u will find out many profitable opportunity. But the problem is transferring balance between two exchanges is very time costuming and end up lossing the arbitrage opportunity.
i searched all over the internet about cross exchange trade that is suitable for arbitrage but no answer, no clue, no starting idea.....

So if anyone, including the developer of this repo, have an idea Please share with me🙏

**Comment by jrbjrb1212** [comment]
Hello,

**Thank you for posting the issue!** 

Great to hear that you are trying to build onto the project. One huge drawback to cross exchange arbitrages is the fact that you have to transfer currencies across exchanges to be able to act on them. Unfortunately because all of these exchanges compete with each other it is hard to transfer currencies between them. Most of the time you have convert to a native currency like USDT and then buy the requested currency on another. This adds so much time to the whole process.

A possible solution is to minimize/eliminate the need to transfer funds across exchanges. Do to this one would need a balance of several currencies on each exchange. Once a cross arbitrage is found market orders can be posted in parallel. The detection algorithm may look for sequential paths like BTC -> ETH -> USDT -> BTC, but all of those trades can happen independent of each other. The problem of transferring between exchanges is why cross exchange arbitrages are not possible to normal market participants. Usually only trading firms are able to take advantage of this market profitability.

To recap here are some things that could solve the problem,
- Keeping a balance of the crypto currencies that will be traded in arbitrages ahead of detections
- Executing market orders of arbitrages in parallel

Hope this helps! Feel free to bounce other ideas off of me
